pkg/apiserver/apiserver.go

```diff
@@ -119,23 +120,17 @@ func NewAPIServer(clientConfig *rest.Config, coreClient kubernetes.Interface, kc
 
 // Run spawns a go routine that exits when apiserver is stopped.
 func (as *APIServer) Run() error {
-	const (
-		retries = 60
-	)
-	go as.server.PrepareRun().Run(as.stopCh)
-
-	for i := 0; i < retries; i++ {
-		ready, err := as.isReady()
+	go func() {
+		err := as.server.PrepareRun().Run(as.stopCh)
 		if err != nil {
-			return fmt.Errorf("checking readiness: %v", err)
+			klog.Warningf("API service stopped: %v", err)
```

Comment:
i think we should pass in log `logr.Logger` into NewAPIServer (similar to other places).

Reply:
yeah, i started to do this and then got annoyed at needing to change the signature of `newServerConfig` and the private methods it calls where we use klog as well. Probably ought to do it though.

Reply:
just need it on *APIServer, right?